integration-tests/src/test_loop/builder.rs

```diff
@@ -76,6 +86,139 @@ pub(crate) struct TestLoopBuilder {
     track_all_shards: bool,
 }
 
+/// Returns true if the chunk should be dropped based on the
+/// `DropCondition::ProtocolUpgradeChunkRange`.
+fn should_drop_chunk_for_protocol_upgrade(
+    epoch_manager_adapter: Arc<dyn EpochManagerAdapter>,
+    chunk: ShardChunkHeader,
+    protocol_version: ProtocolVersion,
+    chunk_ranges: HashMap<ShardUId, std::ops::Range<i64>>,
+) -> bool {
+    let prev_block_hash = chunk.prev_block_hash();
+    let shard_id = chunk.shard_id();
+    let height_created = chunk.height_created();
+    let epoch_id = epoch_manager_adapter.get_epoch_id_from_prev_block(prev_block_hash).unwrap();
+    let shard_layout = epoch_manager_adapter.get_shard_layout(&epoch_id).unwrap();
+    let chunk_shard_uid = ShardUId::from_shard_id_and_layout(shard_id, &shard_layout);
```

Comment:
There is a context that the whole code is applied to the network messages. Which implies that the chunk is new because we don't distribute old chunks :)